design/pipeline-triggers.md

```diff
@@ -0,0 +1,369 @@
+Whenever a PR is raised on the repository, you can choose to trigger your pipeline using PR triggers. PR triggers are not enabled by defaut. You can enable PR triggers on the repository by defining `pr` trigger on the `repository` resource.
+
+
+#### Scenarios
+	I would like to trigger my pipeline only when a PR is targeted to `releases/*` branch of the repository.
+	I would like to trigger my pipeline only when a PR is targeted to the file path `Web/*` of the repository.
+
+
+#### Schema
+```yaml
+resources:         
+  repositories:
+  - repository: string    
+    source: string  
+    pr:        # Optional; pr triggers are disabled by default
+      branches:  # branch conditions to filter the events, optional; Defaults to all branches.
+        include: [ string ]  # branches to consider the trigger events, optional; Defaults to all branches.
+        exclude: [ string ]  # branches to discard the trigger events, optional; Defaults to none.  
+      paths:
+        include: [ string ]  # file paths to consider the trigger events, optional; Defaults to all branches.
+        exclude: [ string ]  # file paths to discard the trigger events, optional; Defaults to none.
+```
+
+#### Examples: PR triggers
+Unless you specify, `pr` triggers are disabled for your repository. You can enable pull request based pipeline runs.
+
+```yaml
+resources:         
+  repositories:
+  - repository: myPHPApp      
+    type: GitHub
+    connection: myGitHubConnection
+    source: ashokirla/phpApp
+    pr: true
+```
+
+You can control the target branches for your pull request based pipeline runs by simple syntax.
+```yaml
+resources:         
+  repositories:
+  - repository: myPHPApp      
+    type: GitHub
+    connection: myGitHubConnection
+    source: ashokirla/phpApp
+    pr: 
+    - master
+    - releases/*
+```
+
+You can specify the branches and file paths to include and exclude.
+```yaml
+resources:         
+  repositories:
+  - repository: myPHPApp      
+    type: GitHub
+    connection: myGitHubConnection
+    source: ashokirla/phpApp
+    pr:
+      branches:
+        include:
+        - releases/*
+        exclude:
+        - releases/experimental/*
+      paths:
+        include:
+        - web/*
+        exclude:
+        - web/README.md
+```
+
+
+
+### Containers
+Whenever a new image got published to the container registry, your pipeline run will be triggered. See [container resource](pipeline-resources.md#resources-containers) for more details.
+
+
+#### Scenarios	
+- I would like to trigger my pipeline whenever a new image got published.
+- I would like to trigger my pipeline whenever a new image got published to ‘East-US’ location (ACR specific filter).
+
+
+#### Schema
+```yaml
+resources:          
+  containers:
+  - container: string       
+    type: enum  
+    connection: string 
+    image: string # container image name, Tag/Digest is optional; defaults to latest image
+    trigger: # Optional; Triggers are enabled by default
```

Comment:
Same as above, this would be a breaking change and there are solid scenarios where triggering isn't desired.

Reply:
We already have a container resource that can be used in a pipeline that does not trigger.  Having this automatically start triggering on existing pipelines one day is going to unexpected behavior for the users.  I would favor an explicit model here and we can certainly have our default getting started experience explicitly define the trigger.

Reply:
Let's definitely discuss it, @ashokirla. I think a reasonable stance would be:
- Because of the scenario, we wish containers automatically came with triggers.
- Because of backcompat, we can't do that.
- In "v0" (unversioned) YAML, container triggers must be explicit. When we rev to v1 (we're going to eventually add a "version: int" field to the pipeline-level), we can change to the behavior we want.

I'll start a list of all the things we want to do with v1 syntax.

Reply:
[Here is that list](https://github.com/Microsoft/azure-pipelines-yaml/pull/92).

Reply:
Note, spinnaker is a *continuous delivery* product. So that's not exactly an apples to apples comparison, since we're building a CI/CD pipeline product.

Reply:
@vtbassmatt : I understand. Given the breaking changes, I think it should be fine if we can update the behavior in v1. As @chrisrpatterson mentioned we should improve the getting started experience.
@ericsciple : Yeah, I agree. But most of the scenarios that involve containers are CD scenarios. And our objective is to build a best in class Container CI/CD solution. So we intend do the right thing automatically.

Reply:
Done changes as suggested.